app/models/user.rb

```diff
@@ -3,4 +3,9 @@ class User < ApplicationRecord
 
   has_many :estimates
   has_many :comments
+
+  # Returns the email if the user doesn't have a name in Github
+  def name
+    read_attribute(:name) || email
```

Comment:
is the email required? what if the email is nil?

Reply:
@JuanVqz Users come from GitHub and we always collect their email address. So there is no way for the email to be nil. 

@FionaDL On the other hand, this doesn't look very straightforward, why do we need the call to `read_attribute`?

I would recommend two different alternatives: 

```suggestion
    name || email
```

Or: 

```suggestion
    name.presence || email
```

Reply:
@etagwerker I tested with your suggestions, but I think it is creating an infinite loop, I get "stack level too deep (SystemStackError)". 

I think we could do self[:name] || email if it feels more straightforward. What do you think?

Reply:
@FionaDL Oh, I understand why that would do that.

How about this instead? 

```
  def name_or_email
    name || email
  end
```

Reply:
I could do that, but then we will have to be sure that name_or_email gets called everywhere that we would want to call `user.name`, right?

Reply:
When it comes to presentation logic, I think it is usually best to have this sort of code in a view helper file. 

So maybe it would be best to move this code to the application_helper file and later change the references in the view to this new method.

So yeah, that means that all view files that call `user.name` will now have to call something like `display_name(user)`